### PR TITLE
Implement material files management

### PR DESCRIPTION
**Pull request by allenwq**
Depends on #505 

This PR also demonstrates how to use the new attachment framework.

### CODE REVIEW

**Comment by lowjoel** [comment]
@allenwq Rebase please.

**Comment by allenwq** [comment]
@lowjoel Updated here.

File: app/views/course/material/folders/_material.slim [hunk 1]
```diff
@@ -0,0 +1,9 @@
+= content_tag_for(:tr, material)
+  th
+    => fa_icon 'file-o'
+    = link_to material.name, [current_course, @folder, material]
+
+  td = format_datetime(material.updated_at)
+  td
+    = edit_button([current_course, @folder, material])
+    = delete_button([current_course, @folder, material])
```
Comment (lowjoel) [inline app/views/course/material/folders/_material.slim]: missing trailing newline

Comment (allenwq) [inline app/views/course/material/folders/_material.slim]: @lowjoel Fixed, Thanks !

**Comment by lowjoel** [comment]
I hope this is mostly following the old design...?

**Comment by allenwq** [comment]
@lowjoel Thanks !

It mostly follows our discussion on the white board, which is different from the old design, especially the controllers and the forms in views.  Models are almost same though.

I think And the major changes are:

1. Materials owners will own the folder instead of materials, so that there is no virtual folders.

2. Because there is no javascript for now, how files are uploaded and associated with their owners have changed.

**Comment by lowjoel** [comment]
Please place the discussion and design in the wiki as documentation.

**Pull request merged**